src/ray/core_worker/core_worker.cc

```diff
@@ -2668,6 +2668,12 @@ void CoreWorker::HandleUpdateObjectLocationBatch(
 
 void CoreWorker::AddObjectLocationOwner(const ObjectID &object_id,
                                         const NodeID &node_id) {
+  if (gcs_client_->Nodes().Get(node_id) == nullptr) {
```

Comment:
Nit: Might be better to explicitly indicate that we're filtering out dead nodes, instead of relying on the default arg,
especially since we're only getting the node info to check if the node is dead.
```suggestion
  if (gcs_client_->Nodes().Get(node_id, /*filter_dead_nodes=*/true) == nullptr) {
```